src/main/kotlin/org/malachite/estella/services/InterviewService.kt

```diff
@@ -1,23 +1,130 @@
+import org.malachite.estella.security.Authority
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
+import java.sql.Timestamp
+import java.util.*
 
 @Service
 class InterviewService(
-    @Autowired private val interviewRepository: InterviewRepository,
-    @Autowired private val mailService: MailService
-) {
+        @Autowired private val interviewRepository: InterviewRepository,
+        @Autowired private val applicationRepository: ApplicationRepository,
+        @Autowired private val hrPartnerService: HrPartnerService,
+        @Autowired private val offerService: OfferService,
+        @Autowired private val mailService: MailService,
+        @Autowired private val recruitmentProcessService: RecruitmentProcessService,
+        @Autowired private val interviewNoteRepository: InterviewNoteRepository,
+        @Autowired private val securityService: SecurityService
+): EStellaService<Interview>() {
+    override val throwable: Exception = InterviewNotFoundException()
 
     fun createInterview(offer: Offer, application: Application, payload: InterviewPayloads = InterviewPayloads()) {
-        val interview = Interview(null, payload.dateTime, payload.minutesLength, application, setOf())
+        val interview = Interview(null, payload.dateTime, payload.minutesLength, application, listOf(), setOf())
         interviewRepository.save(interview)
         mailService.sendInterviewInvitationMail(offer, interview)
     }
 
+    fun createInterview(applicationId: Int, stage: StageType) {
+        val minutesLength: Int = if (stage == StageType.HR_INTERVIEW) 30 else 90
+        val application = applicationRepository.findById(applicationId)
+        application.ifPresent {
+            val interview = Interview(null, null, minutesLength, application.get(), listOf(), setOf())
+            interviewRepository.save(interview)
+            mailService.sendInterviewInvitationMail(offerService.getOfferByApplication(it), interview)
+        }
+    }
+
+
+    fun getInterview(id: UUID): Interview = withExceptionThrower { interviewRepository.findById(id).get() }
+
+    fun getUserFromInterviewUuid(interviewId: UUID): User? =
+        getInterview(interviewId).application.jobSeeker.user
+
+    fun getOrganizationFromPartner(hrPartnerId: Int): Organization =
+            hrPartnerService.getHrPartner(hrPartnerId).organization
+
+    fun getLastInterviewFromApplicationId(applicationId: Int): InterviewId =
+            withExceptionThrower { interviewRepository.getAllByApplicationId(applicationId).sortedWith { a, b ->
+                a.dateTime?.compareTo(b.dateTime) ?: -1
+            }.first() }.toId()
+
+    fun setHosts(id: UUID, hostsMails: List<String>) {
+        if (!canHrUpdate(id)) throw UnauthenticatedException()
+        val interview = interviewRepository.findById(id)
+        interview.ifPresent {
+            val int = interviewRepository.save(it.copy(hosts=hostsMails))
```

Comment:
w javie by nie przeszło kompilacyjnie, tu nie przechodzi przez review. Weź to zmień na jakąś normalną nazwę zmiennej.